From 856371ca1561ca9b3280cc323ff296c7c5e1fa93 Mon Sep 17 00:00:00 2001
From: Pavel Kubelun <be.dissent@gmail.com>
Date: Tue, 22 Nov 2016 17:37:56 +0300
Subject: [PATCH] ipq806x: clk: gcc: add tsens child node

Thermal sensors in ipq806x are inside a Global clock controller.
Add a child node into it to be used by the TSENS driver.

Signed-off-by: Pavel Kubelun <be.dissent@gmail.com>

---
 drivers/clk/qcom/gcc-ipq806x.c | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/clk/qcom/gcc-ipq806x.c
+++ b/drivers/clk/qcom/gcc-ipq806x.c
@@ -3109,6 +3109,7 @@ MODULE_DEVICE_TABLE(of, gcc_ipq806x_matc
 static int gcc_ipq806x_probe(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
+	struct platform_device *tsens;
 	struct regmap *regmap;
 	int ret;
 
@@ -3138,6 +3139,13 @@ static int gcc_ipq806x_probe(struct plat
 	regmap_write(regmap, 0x3cf8, 8);
 	regmap_write(regmap, 0x3d18, 8);
 
+	tsens = platform_device_register_data(&pdev->dev, "qcom-tsens", -1,
+					      NULL, 0);
+	if (IS_ERR(tsens))
+		return PTR_ERR(tsens);
+
+	platform_set_drvdata(pdev, tsens);
+
 	return 0;
 }
 
